### PR TITLE
changes to emonpi lcd emonPiLCD.py to take arguments from config file…

### PR DESCRIPTION
**Pull request by Britnell**
…, that path of which is given as terminal line argument in emonpilcd- service
I broke it then I fixed it.

added to the emonpilcd service file to give config file path
added to emonpilcd.py to take config path as argument
load variables from .conf file instead of having them fixed in python file

Broken for a while, fixed by setting unix filetype in vim, handling path corectly.
then it worked

### CODE REVIEW

File: lcd/emonPiLCD.conf [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+backlight_timeout = 180
+SHUTDOWN_TIME = 3   # seconds   
+GPIO_PORT = P8_11    # >> conf
+GPIO_PORT_shutdown = P8_12    # >> conf
+max_number_pages = 6    # >> conf ??
+
+host = localhost
+port = 6379
+db = 0
+db2 : 1
```
Comment (jlark) [inline lcd/emonPiLCD.conf]: why is this last one a collon `:` on db2

File: lcd/emonPiLCD.py [hunk 2]
```diff
@@ -17,65 +17,190 @@
...
+SHUTDOWN_TIME = get_config('SHUTDOWN_TIME')        #3  # Shutdown after 3 second press
 
 # ------------------------------------------------------------------------------------
 # Start Logging
 # ------------------------------------------------------------------------------------
 import logging
 import logging.handlers
 # NOTE this is during pilots remove this in future or move to proper log management
-uselogfile = True
+uselogfile = True    # >> conf
```
Comment (jlark) [inline lcd/emonPiLCD.py]: did you miss the config for uselogfile?

File: lcd/emonPiLCD.py [hunk 3]
```diff
@@ -17,65 +17,190 @@
...
-mqtt_rx_channel = "emonhub/rx/#"
-mqtt_push_channel = "emonhub/tx/#"
+mqtt_rx_channel =  "emonhub/rx/#"   #get_config('mqtt_rx_channel')   #"emonhub/rx/#"
+mqtt_push_channel = "emonhub/tx/#"    # >> conf
 
 if not uselogfile:
     loghandler = logging.StreamHandler()
 else:
-    loghandler = logging.handlers.RotatingFileHandler("/var/log/emonPiLCD",'a', 5000 * 1024, 1)
+    loghandler = logging.handlers.RotatingFileHandler("/var/log/emonPiLCD",'a', 5000 * 1024, 1)    # >> conf
```
Comment (jlark) [inline lcd/emonPiLCD.py]: also here seems the log location is hardcoded stil?

File: lcd/emonPiLCD.py [hunk 4]
```diff
@@ -17,65 +17,190 @@
...
-uselogfile = True
+uselogfile = True    # >> conf
 
 mqttc = False
 mqttConnected = False
 basedata = []
-mqtt_rx_channel = "emonhub/rx/#"
-mqtt_push_channel = "emonhub/tx/#"
+mqtt_rx_channel =  "emonhub/rx/#"   #get_config('mqtt_rx_channel')   #"emonhub/rx/#"
+mqtt_push_channel = "emonhub/tx/#"    # >> conf
```
Comment (jlark) [inline lcd/emonPiLCD.py]: same here? Shouldn't these be defined in config?

**Comment by jlark** [comment]
hey @Britnell thanks, Please see my comments inline

File: lcd/emonPiLCD.conf (hunk 1 above)
Comment (Britnell) [inline lcd/emonPiLCD.conf]: debugging & testing, removing it

File: lcd/emonPiLCD.py (hunk 2 above)
Comment (Britnell) [inline lcd/emonPiLCD.py]: tick

File: lcd/emonPiLCD.py (hunk 3 above)
Comment (Britnell) [inline lcd/emonPiLCD.py]: tick

File: lcd/emonPiLCD.py (hunk 4 above)
Comment (Britnell) [inline lcd/emonPiLCD.py]: tick.

**Comment by jlark** [comment]
LGTM

**Pull request merged**